src/nvim/api/private/helpers.c

```diff
@@ -971,3 +972,47 @@ static void set_option_value_err(char *key,
     api_set_error(err, Exception, "%s", errmsg);
   }
 }
+
+
+/// Get an array containing dictionaries describing mappings based on mode and buffer id
+///
+/// @param  mode  The abbreviation for the mode
+/// @param  buffer_id  Buffer handle. -1 For global map array
+/// @returns An array of maparg() like dictionaries describing mappings
+ArrayOf(Dictionary) keymap_array(String mode, Integer buffer_id)
```

Comment:
This should accept `buf_T*` (NULL or buf). It is not an API function, you don’t need to restrict types.